lib/util/fs/index.ts

```diff
@@ -11,6 +11,15 @@ export * from './proxies';
 
 export const pipeline = util.promisify(stream.pipeline);
 
+function isPathInBaseDir(path: string, baseDir?: string): boolean {
```

Comment:
Wondering if we can make baseDir mandatory and fix any typings downstream, or is this the best place to address this?

Reply:
`localDir` and `cacheDir` are also optionally null in `RepoGlobalConfig`, which led to this definition. I understand that, if made mandatory, they will need to be supplied in any `RepoGlobalConfig` definition, even if not required because e.g., `cacheDir` is not used.

Reply:
Do I understand right though that if the `baseDir` is ever passed to this function "falsey" (e.g. null, undefined, `''`) then the read or write will fail? i.e. enforcing this with types should push the failure detection earlier ideally, if it's a logic bug of our own doing?

Reply:
If `baseDir` is "falsey", the `if` clause in L15 wouldn't evaluate to true. That means, read/write etc. will work due to `isPathInBaseDir` returning `true`. In other words, if no `baseDir` is provided, the check wouldn't make sense anyway.

If `localDir` or `cacheDir` (and these two ending up in `baseDir`) would be mandatory in `RepoGlobalConfig`, any tests that would need just a minimal config would also require these settings in addition. IMHO, for tests it should be fine to leave out this definition if not needed. Likewise, it should be fine to simply set `localDir = ''`, as is already done. I hope that's want you meant 👍

Reply:
Hopefully at some point we'll have strict non-null types for `localDir` and `cacheDir`, and then we need not to forget restrict this function too